localstack/services/awslambda/invocation/runtime_executor_plugin.py

```diff
@@ -0,0 +1,111 @@
+import logging
+from abc import ABC, abstractmethod
+from typing import Type
+
+from plugin import Plugin, PluginManager
+
+from localstack import config
+from localstack.aws.api.lambda_ import FunctionVersion
+from localstack.services.awslambda.invocation.lambda_models import ServiceEndpoint
+
+LOG = logging.getLogger(__name__)
+
+
+class RuntimeExecutorPlugin(Plugin):
+    namespace = "localstack.lambda.runtime_executor"
+
+
+class RuntimeExecutor(ABC):
+    id: str
+    function_version: FunctionVersion
+
+    def __init__(
+        self, id: str, function_version: FunctionVersion, service_endpoint: ServiceEndpoint
+    ) -> None:
+        """
+        Runtime executor class responsible for executing a runtime in specific environment
+
+        :param id: ID string of the runtime executor
+        :param function_version: Function version to be executed
+        :param service_endpoint: Service endpoint for execution related callbacks
+        """
+        self.id = id
+        self.function_version = function_version
+
+    @abstractmethod
+    def start(self, env_vars: dict[str, str]) -> None:
+        """
+        Start the runtime executor with the given environment variables
+
+        :param env_vars:
+        :return:
+        """
+        pass
+
+    @abstractmethod
+    def stop(self) -> None:
+        """
+        Stop the runtime executor
+        """
+        pass
+
+    @abstractmethod
+    def get_address(self) -> str:
+        """
+        Get the address the runtime executor is available at for the LocalStack container.
+
+        :return: IP address or hostname of the execution environment
+        """
+        pass
+
+    @abstractmethod
+    def get_endpoint_from_executor(self) -> str:
+        """
+        Get the address of LocalStack the runtime execution environment can communicate with LocalStack
+
+        :return: IP address or hostname of LocalStack (from the view of the execution environment)
+        """
+        pass
+
+    @abstractmethod
+    def invoke(self, payload: dict[str, str]) -> None:
```

Comment:
Not sure `dict[str,str]` is the right type here since the payload dict can be nested :thinking:

Reply:
The payload looks like this:
```
        invoke_payload = {
            "invoke-id": invocation_event.invocation_id,
            "invoked-function-arn": invocation_event.invocation.invoked_arn,
            "payload": to_str(invocation_event.invocation.payload),
        }

```
so the payload is just a string here

Reply:
ah ok, was thinking of the payload :+1:  thanks for clearing this up